src/core/cubic_impl.h

```diff
@@ -0,0 +1,106 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
```

Comment:
Why do we need a separate file for this?

Reply:
We need to seperate the definition of `QUIC_CONGESTION_CONTROL_CUBIC` and all cubic functions so that:

1. We can include `cubic.h` in `congestion_control.h` for using `QUIC_CONGESTION_CONTROL_CUBIC` in `union QUIC_CONGESTION_CONTROL_CONTEXT`. This union is to find the max context size for all congestion control algorithms.
2. To avoid circular reference for `cubic_impl.h` and `congestion_control.h`.